test_conformance/api/test_kernel_local_memory_size.cpp

```diff
@@ -0,0 +1,259 @@
+            data[id] += array[i];
+    }
+}
+)CLC";
+
+static const char* local_param_kernel = R"CLC(
+__kernel void local_param_kernel(__local int* local_ptr, __global int* src,
+    __global int* dst) {
+
+    size_t id = get_global_id(0);
+
+    local_ptr[id] = src[id];
+    barrier(CLK_GLOBAL_MEM_FENCE);
+    dst[id] = local_ptr[id];
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id == 9)
+    {
+        for(size_t i = 0; i < 10; i++)
+            dst[id] += local_ptr[i];
+    }
+}
+)CLC";
+
+static const char* local_param_local_memory_kernel = R"CLC(
+__kernel void local_param_local_memory_kernel(__local int* local_ptr,
+    __global int* src, __global int* dst) {
+
+    size_t id = get_global_id(0);
+
+    __local int local_data[10];
+    local_ptr[id] = src[id];
+
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id / 2 == 0) {
+        for(size_t i = 0; i < 10; i++)
+            local_data[id] += local_ptr[i];
+    }
+    else
+        local_data[id] = local_ptr[id] * 2;
+
+    barrier(CLK_LOCAL_MEM_FENCE);
+
+    dst[id] = local_data[id];
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id == 9)
+    {
+        for(size_t i = 0; i < 10; i++)
+            dst[id] += local_data[i];
+        dst[id] += 666;
+    }
+}
+)CLC";
+
+int test_kernel_local_memory_size(cl_device_id deviceID, cl_context context,
+                                  cl_command_queue queue, int num_elements)
+{
+    int error;
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+
+    // Check memory needed to execute empty kernel
+    if (create_single_kernel_helper(context, &program, &kernel, 1,
+                                    &empty_kernel, "empty_kernel")
+        != 0)
+    {
+        return TEST_FAIL;
+    }
+
+    cl_ulong kernel_local_usage = 0;
+    error = clGetKernelWorkGroupInfo(kernel, deviceID, CL_KERNEL_LOCAL_MEM_SIZE,
+                                     sizeof(kernel_local_usage),
+                                     &kernel_local_usage, NULL);
+    test_error(error,
+               "clGetKernelWorkGroupInfo for CL_KERNEL_LOCAL_MEM_SIZE failed");
+
+    test_assert_error(kernel_local_usage >= 0, "kernel local mem size failed");
+
+    // Check memory needed to execute empty kernel with __local variable
+    if (create_single_kernel_helper(context, &program, &kernel, 1,
+                                    &local_memory_kernel, "local_memory_kernel")
+        != 0)
+    {
+        return TEST_FAIL;
+    }
+
+    kernel_local_usage = 0;
+    error = clGetKernelWorkGroupInfo(kernel, deviceID, CL_KERNEL_LOCAL_MEM_SIZE,
+                                     sizeof(kernel_local_usage),
+                                     &kernel_local_usage, NULL);
+    test_error(error,
+               "clGetKernelWorkGroupInfo for CL_KERNEL_LOCAL_MEM_SIZE failed");
+
+    const size_t size = 10;
+    const size_t memory = size * sizeof(cl_int);
+
+    size_t global_work_size[] = { size };
+
+    std::unique_ptr<int[]> data(new int[size]);
```

Comment:
can this be `int data[size]` and make size `constexpr`, I don't think this needs to be allocated on the heap

Reply:
@EwanC done